lib/reports/internal-deadlines/index.js

```diff
@@ -43,13 +43,19 @@ module.exports = ({ db, query: params }) => {
     const q = db.flow('cases')
       .select([
         'cases.*',
-        db.flow.raw('JSON_AGG(activity_log.*) as activity')
+        db.flow.raw('JSON_AGG(activity_log.* ORDER BY activity_log.created_at) as activity')
       ])
       .leftJoin('activity_log', 'cases.id', 'activity_log.case_id')
       .whereRaw(`cases.data->>'model' = 'project'`)
       .whereRaw(`cases.data->>'action' = 'grant'`)
       .where('cases.updated_at', '>', earliest)
       .where('cases.created_at', '<=', latest)
+      .where(orBuilder => {
+        let statuses = [...targetClearingStatuses, ...closedStatuses];
+        return orBuilder.where(withinDateBuilder => withinDateBuilder.where('activity_log.updated_at', '>', earliest).where('activity_log.created_at', '<=', latest))
+          .orWhere(statusBuilder => statusBuilder.whereRaw(`activity_log.event->>'status' in (${statuses.map(_ => '?').join(',')})`, statuses));
```

Comment:
I've never tried this - just out of interest does

```statusBuilder.whereIn(`activity_log.event->>'status'`, statuses)```

not work because it's a json path?

If not I'd be tempted to create a named var just so it's immediately obvious this is parameterising the query

```const placeholders = statuses.map(_ => '?').join(',');```